### DISCUSSION TITLE
Is there a way to dynamically validate swagger ui examples

### DISCUSSION THREAD

**Discussion by vignesh14052002** [fastapi/fastapi#12244] [Questions]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
no code needed for this question
```

### Description

Dynamically generating swagger ui based on api definitions in code is a great feature, on similar lines, is it possible to validate all the endpoints and example payloads in the swagger UI is always in working state (maybe a pytest test case that validates using the OpenAPI.json)?. 

### Operating System

Windows

### FastAPI Version

0.104.1

### Pydantic Version

2.5.0

### Python Version

3.11.3

**Comment by hmol** [comment]
I might not fully understand what you are after here. But if the swagger always are generated from the code, why not just test the code?

**Comment by ceb10n** [comment]
As @hmol mentioned, openapi is generated from code, so testing the code is the easiest way to validate.

I'm not sure if testing the openapi generation part is a good idea, since it's fastapi's responsibility to test, not your application.

Now if you want to generate rules to your APIs, check if they are following the standards you want, etc., you can use [spectral](https://docs.stoplight.io/docs/spectral/4dec24461f3af-open-api-rules) on you ci. For example, you can validate if all your services are using `JSON:API`,  `HAL`, etc...
